### PR TITLE
fix: handle non-2xx status codes returned in provider http requests

### PR DESCRIPTION
**Pull request by kangmingtay**

### CODE REVIEW

**Review by dragarcia — APPROVED** [review]

File: api/provider/errors.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+// Cause returns the root cause error
+func (e *HTTPError) Cause() error {
+	if e.InternalError != nil {
+		return e.InternalError
+	}
+	return e
+}
+
```
Comment (J0) [inline api/provider/errors.go:31]: bit late but TIL: Go does not have ternaries

File: api/provider/notion.go [hunk 2]
```diff
@@ -83,6 +84,10 @@ func (g notionProvider) GetUserData(ctx context.Context, tok *oauth2.Token) (*Us
 		return nil, err
 	}
 
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("a %v error occured with retrieving user from notion", resp.StatusCode)
```
Comment (J0) [inline api/provider/notion.go:88]: /nit
```suggestion
		return nil, fmt.Errorf("a %v error occurred with retrieving user from notion", resp.StatusCode)
```

File: api/provider/twitch.go [hunk 3]
```diff
@@ -98,6 +99,10 @@ func (t twitchProvider) GetUserData(ctx context.Context, tok *oauth2.Token) (*Us
 		return nil, err
 	}
 
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("a %v error occured with retrieving user from twitch", resp.StatusCode)
+	}
```
Comment (J0) [inline api/provider/twitch.go:104]: /nit
```suggestion
		return nil, fmt.Errorf("a %v error occurred with retrieving user from notion", resp.StatusCode)
```

**Review by J0 — APPROVED** [review]
couple of minor/inconsequential nit, LGTM

**Pull request merged**